### ISSUE TITLE
we send presence updates to all the servers that our users share rooms with, whenever a remote user joins any of those rooms

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#3962]
I suspect this is the root cause of #2514 and a bunch of other similar reports like #1429 and #3490.

Whenever a remote user joins a room, we look for any local users that are in that room, and then, for each local user, send out presence updates to *all* the servers that user shares a room with.

**Comment by jevolk** [comment]
This should be a high priority "p1" or whatever issue. The presence spam is a rather large `N*N` load on the federation which will return considerable value after investing in its redress. It also has security implications leaking usernames to servers.

Here's an example just looking at [Construct](https://github.com/matrix-construct/construct)'s console just _this moment_:
```
1540156656.160850     INFO      ircd   client     44 :4ray.co @jean:4ray.co is active and online 15 seconds ago
1540156657.311620     INFO      ircd   client     94 :4ray.co @jean:4ray.co is active and online 17 seconds ago
```
And there are a plethora of others which are immediately discarded as redundant or stale and don't even show up.

**Comment by richvdh** [comment]
this needs fixing pre-v1.0.

**Comment by Half-Shot** [comment]
Anecdotally, I'm now seeing outgoing federation transactions at the rate of about 70Hz which is causing my server to fall behind sending PDUs, which in turn makes conversations quite difficult to have. If we can't fix the spam, could the federation sender at least prioritise PDUs over EDUs?

**Comment by richvdh** [comment]
sure, feel free to write a PR. 

frankly it should be easier to fix the bug.

**Comment by erikjohnston** [comment]
Hopefully fixed by #4942
